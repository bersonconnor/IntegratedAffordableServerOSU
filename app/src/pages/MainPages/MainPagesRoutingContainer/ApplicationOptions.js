import React, {Component} from "react";
import Nav from "react-bootstrap/Nav";
import Collapse from "react-bootstrap/Collapse";
import {LinkContainer} from "react-router-bootstrap";
import "./scss/sidebar.scss";

class ApplicationOptions extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      open_app: this.props.open_app
    };
  }

  render() {
    const { open_app } = this.state;
    const appCenterPaths = [
      /\/manager*/,
      /\/apply*/,
      /\/message*/,
      /\/search*/
    ];
    const appCenterIsActive = appCenterPaths.reduce((truth, curr) => {
      return curr.test(window.location.pathname) || truth;
    }, false);

    return (
      <div>
        {/* <LinkContainer to="/application">
          <Nav.Link
            onClick={() =>
              this.setState({
                open_app: !open_app
              })
            }
            aria-controls="app-center-navigation"
            aria-expanded={open_app}
            className={appCenterIsActive ? "active" : ""}
          >
            Application Center
          </Nav.Link>
        </LinkContainer> */}
        {/* <Collapse in={this.state.open_app} id="app-center-navigation">
          <div id="app-center-navigation">
            <div className="row">
              <div className="col-2"></div>
              <div className="col">
                <Nav
                  className="flex-column"
                  activeKey={window.location.pathname}
                > */}
                  <LinkContainer to="/apply">
                    <Nav.Link>Apply</Nav.Link>
                  </LinkContainer>
                  <LinkContainer to="/search">
                    <Nav.Link>Search</Nav.Link>
                  </LinkContainer>
                {/* </Nav>
              </div>
            </div>
          </div>
        </Collapse> */}
      </div>
    );
  }
}

export default ApplicationOptions;
