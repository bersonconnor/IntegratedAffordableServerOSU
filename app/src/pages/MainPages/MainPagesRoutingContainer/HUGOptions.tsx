import React, { Component } from "react";
import Nav from "react-bootstrap/Nav";
import Collapse from "react-bootstrap/Collapse";
import { LinkContainer } from "react-router-bootstrap";
import "./scss/sidebar.scss";

class HUGOptions extends Component<{}, {}> {
  constructor(props, context) {
    super(props, context);

    // Note: I don't think this component actually does much yet.
    // The state was stored within itself before I got here, so I'm just confused.
    // Only one prop was accessed, but no component actually used the prop.
  }

  render() {
    const HUGPaths = [/\/search*/, /\/hug*/, /\/manage*/, /\/analytics*/];
    const HUGIsActive = HUGPaths.reduce((truth, curr) => {
      return curr.test(window.location.pathname) || truth;
    }, false);

    return (
      <div>
        <LinkContainer to="/hug">
          <Nav.Link
            onClick={() => console.log("HUGOptions navlink clicked")}
            aria-controls="HUG-navigation"
            // aria-expanded={open_HUG}
            className={HUGIsActive ? "active" : ""}
          >
            {">"} Health Utilizing Grants
          </Nav.Link>
        </LinkContainer>
        {/*<Collapse in={this.state.open_HUG} id="HUG-navigation">*/}
        <div id="HUG-navigation">
          <div className="row">
            <div className="col-2"></div>
            <div className="col">
              <Nav className="flex-column" activeKey={window.location.pathname}>
                <LinkContainer to="/search">
                  <Nav.Link>Search</Nav.Link>
                </LinkContainer>
                <LinkContainer to="/hug">
                  <Nav.Link>Create</Nav.Link>
                </LinkContainer>
                <LinkContainer to="/manage">
                  <Nav.Link>Manage</Nav.Link>
                </LinkContainer>
                <LinkContainer to="/analytics">
                  <Nav.Link>Analytics</Nav.Link>
                </LinkContainer>
              </Nav>
            </div>
          </div>
        </div>
        {/*</Collapse>*/}
      </div>
    );
  }
}

export default HUGOptions;
