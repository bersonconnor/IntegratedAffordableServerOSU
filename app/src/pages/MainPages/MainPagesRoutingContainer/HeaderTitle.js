import React, { Component } from "react";
import { withRouter } from "react-router-dom";
// Create your own sass file :D

class HeaderTitle extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    let pageName = "";
    const location = window.location.pathname;
    if (
      location.includes("affiliation") ||
      location.includes("organization") ||
      location.includes("manager")
    ) {
      pageName = "Organization Center";
    } else if (location.includes("application")) {
      pageName = "Application Center";
    } else if (location.includes("hug")) {
      pageName = "HUG Center";
    } else if (location.includes("search")) {
      pageName = "HUG Search";
    } else if (location.includes("manage")) {
      pageName = "HUG Manager";
    } else if (location.includes("dashboard")) {
      pageName = "Dashboard";
    } else if (location.includes("settings")) {
      pageName = "Settings";
    } else if (location.includes("profile")) {
      pageName = "Profile";
    } else if (location.includes("support")) {
      pageName = "Support";
    } else if (location.includes("mdm")) {
      pageName = "Medical Debt Marketplace";
    }
    return (
      <div className="col">
        <h1 className="main-header">{pageName}</h1>
      </div>
    );
  }
}

export default HeaderTitle;
