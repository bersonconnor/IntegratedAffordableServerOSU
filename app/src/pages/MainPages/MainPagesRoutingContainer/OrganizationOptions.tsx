import React, {Component} from "react";
import Nav from "react-bootstrap/Nav";
import Collapse from "react-bootstrap/Collapse";
import {LinkContainer} from "react-router-bootstrap";

import "./scss/sidebar.scss";

class OrganizationOptionsProps {
  userId: number;
  openApp?: any;
  openOrg?: any;
}

class OrganizationOptionsState {
  openApp?: any;
  openOrg?: any;
}

class OrganizationOptions extends Component<
  OrganizationOptionsProps,
  OrganizationOptionsState
> {
  constructor(props, context) {
    super(props, context);

    this.state = {
      openApp: this.props.openApp,
      openOrg: this.props.openOrg
    };
  }

  render() {
    const { openOrg } = this.state;
    const organizationPaths = [
      /\/affiliation*/,
      /\/manager*/,
      /\/services*/,
      /\/analytics*/
    ];
    const organizationIsActive = organizationPaths.reduce((truth, curr) => {
      return curr.test(window.location.pathname) || truth;
    }, false);

    return (
      <div>
        <LinkContainer to="/affiliation">
          <Nav.Link
            onClick={() =>
              this.setState({
                openOrg: !openOrg
              })
            }
            aria-controls="organization-navigation"
            aria-expanded={openOrg}
            className={organizationIsActive ? "active" : ""}
          >
            Organizations
          </Nav.Link>
        </LinkContainer>
        <Collapse in={this.state.openOrg} key="organization-navigation">
          <div id="organization-navigation">
            <div className="row">
              <div className="col-2"></div>
              <div className="col">
                <Nav
                  className="flex-column"
                  activeKey={window.location.pathname}
                >
                  <LinkContainer to="/affiliation">
                    <Nav.Link>Affiliation</Nav.Link>
                  </LinkContainer>
                  <LinkContainer to="/manager">
                    <Nav.Link>Manager</Nav.Link>
                  </LinkContainer>
                  <LinkContainer to="/services">
                    <Nav.Link>Services</Nav.Link>
                  </LinkContainer>
                  <LinkContainer to="/analytics">
                    <Nav.Link>Analytics</Nav.Link>
                  </LinkContainer>
                </Nav>
              </div>
            </div>
          </div>
        </Collapse>
      </div>
    );
  }
}

export default OrganizationOptions;
