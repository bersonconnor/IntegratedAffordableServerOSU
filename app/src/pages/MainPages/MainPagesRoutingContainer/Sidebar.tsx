import React, { Component } from "react";
import Nav from "react-bootstrap/Nav";
import { LinkContainer } from "react-router-bootstrap";
import ApplicationOptions from "./ApplicationOptions";
import OrganizationOptions from "./OrganizationOptions";
import HUGOptions from "./HUGOptions";
import RepresentingOrganization from "./RepresentingOrganization";
import "./scss/sidebar.scss";
import logo from "./Logo.png";
import { OrganizationMembership, UserType } from "affordable-shared-models";
import { AffordableClient } from "affordable-client";
import DepositButton from "../../../components/Modal/DepositButton";
import WithdrawButton from "../../../components/Modal/WithdrawButton";


interface SidebarProps {
  client: AffordableClient;
  userId: number;
  user: string;
  userName: string;
  userType: UserType;
  organizationMembership: OrganizationMembership;
  updateOrganizationFn: (organization: OrganizationMembership) => void;
}

class SidebarState {
  pendingBalance: number = 0;
  availableBalance: number = 0;
  name: any;
  checkBalance: any;
}

class Sidebar extends Component<SidebarProps, SidebarState> {
  constructor(props, context) {
    super(props, context);

    this.state = new SidebarState();
    this.handleChange = this.handleChange.bind(this);
    this.updateSelectedOrganization = this.updateSelectedOrganization.bind(
      this
    );
  }

  async componentDidMount() {
    var timer = setInterval(() => this.getBalance(), 30000);
    this.setState({
      checkBalance: timer
    });
  }

  componentWillUnmount() {
    clearInterval(this.state.checkBalance);
  }


  //for handling verified orgs
  updateSelectedOrganization(organization: OrganizationMembership): void {
    this.props.updateOrganizationFn(organization);
  }

  // Use the submitted data to set the state
  handleChange(event): void {
    const { name, value } = event.target;
    // this.setState({
    //   [name]: value
    // });
  }

  handleSubmit = e => {
    e.preventDefault();
  };

  formatBalance(pending, available) {
    var p = pending.toLocaleString(undefined, { minimumFractionDigits: 2 });
    var a = available.toLocaleString(undefined, { minimumFractionDigits: 2 });
    this.setState({
      pendingBalance: p,
      availableBalance: a
    });
  }

  getBalance = async () => {
    var balance = await this.props.client.getBalance(this.props.user, this.props.userType);
    if (balance.success === "Balance found") {
      this.formatBalance(balance.pendingBalance, balance.balance);
    }
  }


  render() {
    return (
      <div className="col-3 px-0 sidebar" onSubmit={this.handleSubmit}>
        <div className="row mt-3">
          <div className="col text-center">
            <img
              className="format-logo-sidebar"
              src={logo}
              alt="Affordable Logo"
            />
          </div>
        </div>
        {/*representing organization restrictions here*/}
        <div className="row mt-3">
          <div className="col-2" />
          <div className="col">
            {this.props.userType === UserType.RECIPIENT ? (
              <div />
            ) : (
                <RepresentingOrganization
                  client={this.props.client}
                  userId={this.props.userId}
                  representedOrganization={this.props.organizationMembership}
                  updateOrganizationFn={this.updateSelectedOrganization.bind(
                    this
                  )}
                />
              )}
          </div>
        </div>

        <div className="row mt-3">
          <div className="col-2" />
          <div className="col">
            <Nav className="flex-column">
              {"admin".localeCompare(this.props.user) === 0 ?
                <LinkContainer to="/dashboard">
                  <Nav.Link>Dashboard</Nav.Link>
                </LinkContainer>
                : this.props.userType === UserType.RECIPIENT ?

                  <LinkContainer to="/application">
                    <Nav.Link>Dashboard</Nav.Link>
                  </LinkContainer>
                  :
                  <LinkContainer to="/dashboard">
                    <Nav.Link>Dashboard</Nav.Link>
                  </LinkContainer>
              }
              {/*
              <LinkContainer to="/profile">
                <Nav.Link>Profile</Nav.Link>
              </LinkContainer>
              */}
              <LinkContainer to="/transactions">
                <Nav.Link>Transactions</Nav.Link>
              </LinkContainer>
              <LinkContainer to="/settings">
                <Nav.Link>Settings</Nav.Link>
              </LinkContainer>
		
		<LinkContainer to="/filemanager">
			<Nav.Link>File Manager</Nav.Link>
		</LinkContainer>

              {/* integration options shown for donor*/}
              {/* this.props.userType === UserType.DONOR ? (
                <LinkContainer to="/integrate">
                  <Nav.Link>Integrate</Nav.Link>
                </LinkContainer>
              ) : (
                <div />
              )} */}

              {/*application vs organization options for recipient vs donor*/}
              {/* {this.props.userType === UserType.RECIPIENT ? (
                <div>
                  <ApplicationOptions />
                </div>
              ) : (
                <div>
                  <OrganizationOptions userId={this.props.userId} />
                </div>
              )} */}
              {/*hug options present or hidden for donor vs recipient*/}
              {/* {this.props.userType === UserType.DONOR &&
              this.props.organizationMembership?.organization?.isVerified ? (
                <div>
                  <HUGOptions />
                </div>
              ) : (
                <div />
              )} */}
              <a className="nav-link" href="mailto:support@affordhealth.org">SUPPORT</a>
            </Nav>

            {/* User balance and button container */}
            {/* (Don't show if the user is the admin) */}
            {
              this.props.userName === "admin" ?
                <div />
                :
                <div className="col" >
                  <br />
                  {/* User balance */}
                        Balance
                        <br />
                  <div id="sidebar-balance">
                    <span text-align="left">${this.state.availableBalance}</span>
                  </div>

                  {/* A "Withdraw" button for recipient users or "Deposit" for donor users */}
                  {
                    this.props.userType === UserType.DONOR ?
                      <DepositButton />
                      :
                      <WithdrawButton />
                  }
                </div>
            }
          </div>
        </div>
      </div>
    );
  }
}

export default Sidebar;
