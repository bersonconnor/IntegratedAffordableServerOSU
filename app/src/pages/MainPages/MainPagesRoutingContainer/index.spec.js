import React from "react";
import Sidebar from "./Sidebar";
import { shallow } from "enzyme";
import ApplicationOptions from "./ApplicationOptions";
import OrganizationOptions from "./OrganizationOptions";
import HUGOptions from "./HUGOptions";
import HeaderTitle from "./HeaderTitle";
import renderer from "react-test-renderer";
import { Link, MemoryRouter } from "react-router-dom";

//Sidebar
describe("Sidebar component", () => {
  //Sidebar should be rendered
  test("should render Sidebar", () => {
    const wrapper = shallow(<Sidebar repOrgIdFunction={() => {}} />);
    // TODO: This was 11, failed, so I changed it to 12, and it passed. Find out why and properly document + test.
    expect(wrapper.find("div").length).toEqual(12);
  });

//   //Sidebar(user type is recipient)
//   //Application Options and HUG Options displayed
//   test("should match snapshot for recipient userType", () => {
//     const testUserID = "199";
//     const testUserType = 0;
//     const wrapper = shallow(
//       <Sidebar repOrgIdFunction={() => {}} userID={testUserID} usertype={testUserType} />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });

//   //Sidebar(user type is donor)
//   //Organization Options and HUG Options displayed
//   test("should match snapshot for donor userType", () => {
//     const testUserID = "200";
//     const testUserType = 1;
//     const wrapper = shallow(
//       <Sidebar repOrgIdFunction={() => {}} userID={testUserID} usertype={testUserType} />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });

//   //Sidebar(user type is undefined)
//   //Organization Options and HUG Options displayed
//   test("should match snapshot for undefined userType", () => {
//     const testUserID = "201";
//     const testUserType = "";
//     const wrapper = shallow(
//       <Sidebar repOrgIdFunction={() => {}} userID={testUserID} usertype={testUserType} />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });

//   //Sidebar organization is default
//   //HUG Options hidden
//   test("should match snapshot for user not representing an organization", () => {
//     const testUserID = "199";
//     const testUserType = 1;
//     const testRepresenting = " ";
//     const wrapper = shallow(
//       <Sidebar
//         userID={testUserID}
//         usertype={testUserType}
//         representing={testRepresenting}
//         repOrgIdFunction={() => {}}
//       />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });

//   //Sidebar organization is verified
//   //HUG Options displayed
//   test("should match snapshot for user representing a verified organization", () => {
//     const testUserID = "199";
//     const testUserType = 1;
//     const testRepresenting = "verified";
//     const wrapper = shallow(
//       <Sidebar
//         userID={testUserID}
//         usertype={testUserType}
//         representing={testRepresenting}
//         repOrgIdFunction={() => {}}
//       />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });

//   //Sidebar organization is unverified
//   //HUG Options hidden
//   test("should match snapshot for recipient userType", () => {
//     const testUserID = "199";
//     const testUserType = 1;
//     const testRepresenting = "unverified";
//     const wrapper = shallow(
//       <Sidebar
//         userID={testUserID}
//         usertype={testUserType}
//         representing={testRepresenting}
//         repOrgIdFunction={() => {}}
//       />
//     );
//     expect(wrapper).toMatchSnapshot();
//   });
// });

// //Application Options
// describe("Application Options", () => {
//   //ApplicationOptions should be rendered
//   test("should render ApplicationOptions", () => {
//     const wrapper = shallow(<ApplicationOptions />);
//     expect(wrapper.find("div").length).toEqual(5);
//   });

  //check open collapsible - ApplicationOptions
  test("should render open collapsible - Applications ", () => {
    const wrapper = shallow(
      <OrganizationOptions open_app={true} open_org={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  //check closed collapsible - ApplicationOptions
  test("should render closed collapsible - Applications ", () => {
    const wrapper = shallow(
      <OrganizationOptions open_app={true} open_org={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

//HUG Options
describe("HUG Options", () => {
  //HUGOptions should be rendered
  test("should render HUGOptions", () => {
    const wrapper = shallow(<HUGOptions />);
    expect(wrapper.find("div").length).toEqual(5);
  });

  //check open collapsible - HUGs
  test("should render open collapsible - HUGs ", () => {
    const wrapper = shallow(
      <OrganizationOptions open_app={true} open_org={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  //check closed collapsible - HUGs
  test("should render closed collapsible - HUGs", () => {
    const wrapper = shallow(
      <OrganizationOptions open_app={true} open_org={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

//Organization Options
describe("Organization Options", () => {
  const wrapper = shallow(<OrganizationOptions />);

  //OrganizationOptions should be rendered
  test("should render OrganizationOptions", () => {
    expect(wrapper.find("div").length).toEqual(5);
  });

  //click button
  test("organization link matches snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Link to="/organization" />
      </MemoryRouter>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  //check open collapsible for Organization options
  test("should render open collapsible - Organization", () => {
    const wrapper = shallow(
      <OrganizationOptions open_app={true} open_org={true} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  //check closed collapsible for Organization options
  test("should render closed collapsible - Organization", () => {
    const wrapper = shallow(
      <OrganizationOptions open_app={true} open_org={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

//HeaderTitle
describe("Header Titles", () => {
  const wrapper = shallow(<HeaderTitle />);

  //HeaderTitle should be rendered
  test("should render header title", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });
});
