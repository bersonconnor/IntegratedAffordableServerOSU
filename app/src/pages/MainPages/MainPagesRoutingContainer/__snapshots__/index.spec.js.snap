// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HUG Options should render closed collapsible - HUGs 1`] = `
<div>
  <withRouter(LinkContainer)
    to="/affiliation"
  >
    <NavLink
      aria-controls="organization-navigation"
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      className=""
      disabled={false}
      onClick={[Function]}
    >
      Organizations
    </NavLink>
  </withRouter(LinkContainer)>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    key="organization-navigation"
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="organization-navigation"
    >
      <div
        className="row"
      >
        <div
          className="col-2"
        />
        <div
          className="col"
        >
          <Nav
            activeKey="/"
            className="flex-column"
            fill={false}
            justify={false}
          >
            <withRouter(LinkContainer)
              to="/affiliation"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Affiliation
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/manager"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Manager
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/services"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Services
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/analytics"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Analytics
              </NavLink>
            </withRouter(LinkContainer)>
          </Nav>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`HUG Options should render open collapsible - HUGs  1`] = `
<div>
  <withRouter(LinkContainer)
    to="/affiliation"
  >
    <NavLink
      aria-controls="organization-navigation"
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      className=""
      disabled={false}
      onClick={[Function]}
    >
      Organizations
    </NavLink>
  </withRouter(LinkContainer)>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    key="organization-navigation"
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="organization-navigation"
    >
      <div
        className="row"
      >
        <div
          className="col-2"
        />
        <div
          className="col"
        >
          <Nav
            activeKey="/"
            className="flex-column"
            fill={false}
            justify={false}
          >
            <withRouter(LinkContainer)
              to="/affiliation"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Affiliation
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/manager"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Manager
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/services"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Services
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/analytics"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Analytics
              </NavLink>
            </withRouter(LinkContainer)>
          </Nav>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`Organization Options organization link matches snapshot 1`] = `
<a
  href="/organization"
  onClick={[Function]}
/>
`;

exports[`Organization Options should render closed collapsible - Organization 1`] = `
<div>
  <withRouter(LinkContainer)
    to="/affiliation"
  >
    <NavLink
      aria-controls="organization-navigation"
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      className=""
      disabled={false}
      onClick={[Function]}
    >
      Organizations
    </NavLink>
  </withRouter(LinkContainer)>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    key="organization-navigation"
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="organization-navigation"
    >
      <div
        className="row"
      >
        <div
          className="col-2"
        />
        <div
          className="col"
        >
          <Nav
            activeKey="/"
            className="flex-column"
            fill={false}
            justify={false}
          >
            <withRouter(LinkContainer)
              to="/affiliation"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Affiliation
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/manager"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Manager
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/services"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Services
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/analytics"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Analytics
              </NavLink>
            </withRouter(LinkContainer)>
          </Nav>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`Organization Options should render open collapsible - Organization 1`] = `
<div>
  <withRouter(LinkContainer)
    to="/affiliation"
  >
    <NavLink
      aria-controls="organization-navigation"
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      className=""
      disabled={false}
      onClick={[Function]}
    >
      Organizations
    </NavLink>
  </withRouter(LinkContainer)>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    key="organization-navigation"
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="organization-navigation"
    >
      <div
        className="row"
      >
        <div
          className="col-2"
        />
        <div
          className="col"
        >
          <Nav
            activeKey="/"
            className="flex-column"
            fill={false}
            justify={false}
          >
            <withRouter(LinkContainer)
              to="/affiliation"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Affiliation
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/manager"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Manager
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/services"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Services
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/analytics"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Analytics
              </NavLink>
            </withRouter(LinkContainer)>
          </Nav>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`Sidebar component should render closed collapsible - Applications  1`] = `
<div>
  <withRouter(LinkContainer)
    to="/affiliation"
  >
    <NavLink
      aria-controls="organization-navigation"
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      className=""
      disabled={false}
      onClick={[Function]}
    >
      Organizations
    </NavLink>
  </withRouter(LinkContainer)>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    key="organization-navigation"
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="organization-navigation"
    >
      <div
        className="row"
      >
        <div
          className="col-2"
        />
        <div
          className="col"
        >
          <Nav
            activeKey="/"
            className="flex-column"
            fill={false}
            justify={false}
          >
            <withRouter(LinkContainer)
              to="/affiliation"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Affiliation
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/manager"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Manager
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/services"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Services
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/analytics"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Analytics
              </NavLink>
            </withRouter(LinkContainer)>
          </Nav>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;

exports[`Sidebar component should render open collapsible - Applications  1`] = `
<div>
  <withRouter(LinkContainer)
    to="/affiliation"
  >
    <NavLink
      aria-controls="organization-navigation"
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      className=""
      disabled={false}
      onClick={[Function]}
    >
      Organizations
    </NavLink>
  </withRouter(LinkContainer)>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    key="organization-navigation"
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="organization-navigation"
    >
      <div
        className="row"
      >
        <div
          className="col-2"
        />
        <div
          className="col"
        >
          <Nav
            activeKey="/"
            className="flex-column"
            fill={false}
            justify={false}
          >
            <withRouter(LinkContainer)
              to="/affiliation"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Affiliation
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/manager"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Manager
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/services"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Services
              </NavLink>
            </withRouter(LinkContainer)>
            <withRouter(LinkContainer)
              to="/analytics"
            >
              <NavLink
                as={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "SafeAnchor",
                    "render": [Function],
                  }
                }
                disabled={false}
              >
                Analytics
              </NavLink>
            </withRouter(LinkContainer)>
          </Nav>
        </div>
      </div>
    </div>
  </Collapse>
</div>
`;
